specification/schemaregistry/SchemaRegistry/main.tsp

```diff
@@ -0,0 +1,209 @@
+  #suppress "@azure-tools/typespec-azure-core/use-standard-operations" "Standard createOrReplace operation does not allow passing in binary body to request."
+  @summary("Register new schema")
+  @doc("Register new schema. If schema of specified name does not exist in specified group, schema is created at version 1. If schema of specified name exists already in specified group, schema is created at latest version + 1.")
+  @route("/$schemaGroups/{groupName}/schemas/{name}")
+  @put
+  op registerSchema is Azure.Core.Foundations.Operation<
+    Azure.Core.Foundations.ItemKeysOf<Schema> & SchemaContent & SchemaContentType,
+    NoContentResponse<SchemaHeaders>
+  >;
+}
+
+// Models
+
+alias NoContentResponse<T> = TypeSpec.Http.Response<204> & T;
+
+@doc("The content type for given schema. Each schema type has an associated content-type.")
+model SchemaContentType {
+  @doc("The content type for given schema.")
+  @header("content-type")
+  contentType: "application/json; serialization=Avro" | "application/json; serialization=json" | "text/plain; charset=utf-8";
+}
+
+@doc("Content of the schema.")
+model SchemaContent {
+  @body
+  @doc("String representation (UTF-8) of the schema.")
+  content: bytes;
+}
+
+@doc("No Content.")
+model SchemaHeaders {
+  @doc("URL location of schema, identified by schema group, schema name, and version.")
+  @header
+  location: string;
+
+  @doc("References specific schema in registry namespace.")
+  @header
+  schemaId: string;
+
+  @doc("URL location of schema, identified by schema ID.")
+  @header
+  schemaIdLocation: string;
+
+  @doc("References schema group.")
+  @header
+  schemaGroupName: string;
+
+  @doc("References schema name.")
+  @header
+  schemaName: string;
+
+  @doc("Version of the returned schema.")
+  @header
+  schemaVersion: int32;
+}
+
+@doc("Object received from the registry containing schema identifiers.")
+@resource("$schemaGroups/$schemas")
+model SchemaId {
+  @key("id")
+  @doc("Schema ID that uniquely identifies a schema in the registry namespace.")
+  @visibility("read")
+  id: string;
+}
+
+@doc("Schema Group resource.")
+@resource("$schemaGroups")
+model SchemaGroup {
+  @key("groupName")
+  @doc("Name of schema group.")
+  @visibility("read")
+  groupName: string;
+}
+
+@doc("Schemas resource.")
+@resource("schemas")
+@parentResource(SchemaGroup)
+model Schema {
+  @key("name")
+  @doc("Name of schema.")
+  @visibility("read")
+  @maxLength(50)
+  @pattern("^[A-Za-z0-9][^\\\\/$:]*$")
+  @extension("x-ms-parameter-location", "method")
+  name: string;
+
+  @doc("String representation (UTF-8) of the registered schema.")
+  schemaContent: bytes;
+}
+
+@doc("Schema versions resource.")
+@resource("versions")
+@parentResource(Schema)
+model Version {
+  @key("schemaVersion")
+  @doc("Version number of specific schema.")
+  @visibility("read")
+  @extension("x-ms-parameter-location", "method")
```

Comment:
Again, if you dont include this does the SDK not generate correctly?

Reply:
it works!